### PR TITLE
Add example local environment with traefik

### PR DESCRIPTION
**Pull request by fcollonval**
## Description

Add example of Traefik with Keycloak to demonstrate the new documented configuration using `static://202`

## Motivation and Context

Make life easier for interested reader by providing a traefik example.

## How Has This Been Tested?

I run it locally.

## Checklist:

- [x] ~~My change requires a change to the documentation or CHANGELOG.~~
- [ ] I have updated the documentation/CHANGELOG accordingly.
- [x] I have created a feature (non-master) branch for my PR.

### CODE REVIEW

**Comment by NickMeves** [comment]
CC: @linuxgemini 

I have no clue for Traefik stuff to leave worthwhile feedback 😅

**Comment by linuxgemini** [comment]
Although its looking good to me, I'm out of touch on Keycloak so this PR has to be reviewed once more.

File: contrib/local-environment/oauth2-proxy-traefik.cfg [hunk 1]
```diff
@@ -0,0 +1,23 @@
+http_address="0.0.0.0:4180"
+cookie_secret="OQINaROshtE9TcZkNAm-5Zs2Pv3xaWytBmc5W7sPX7w="
+email_domains=["example.com"]
+cookie_secure="false"
+# upstreams="http://httpbin"
```
Comment (JoelSpeed) [inline contrib/local-environment/oauth2-proxy-traefik.cfg:5]: Should probably remove this before merging

File: contrib/local-environment/oauth2-proxy-traefik.cfg [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+# in this case oauth2-proxy is going to visit
+# http://keycloak.localtest.me:9080/auth/realms/master/.well-known/openid-configuration for configuration
+oidc_issuer_url="http://keycloak.localtest.me/auth/realms/master"
+provider="oidc"
+provider_display_name="Keycloak"
+
+reverse_proxy="true"
+
+skip_provider_button="true"
```
Comment (JoelSpeed) [inline contrib/local-environment/oauth2-proxy-traefik.cfg:23]: I think these are new right? Perhaps some comments why these are here would be good for future users

**Review by JoelSpeed — COMMENTED** [review]
I'm also not a Traefik user so can't give much feedback there, though it seems ok.

Since most of our other examples use Dex (and the config is simpler with Dex), just wondering why you went for Keycloak in this example and whether we should include a Dex example instead/as well, WDYT?

I'll try and give it run on my machine later and check it all out, if that passes, I'm pretty happy to merge

File: contrib/local-environment/oauth2-proxy-traefik.cfg (hunk 1 above)
Comment (fcollonval) [reply contrib/local-environment/oauth2-proxy-traefik.cfg]: :+1:

File: contrib/local-environment/oauth2-proxy-traefik.cfg (hunk 2 above)
Comment (fcollonval) [reply contrib/local-environment/oauth2-proxy-traefik.cfg]: Indeed this is possible thanks to the new documented `ForwardAuth` + static upstream configuration.

I added a comment.

**Comment by fcollonval** [comment]
Hey thanks for reviewing. I updated the example to use dex and to be as close as possible to the nginx example except that it uses traefik.

File: contrib/local-environment/docker-compose-traefik.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
+# This docker-compose file can be used to bring up an example instance of oauth2-proxy
+# for manual testing and exploration of features.
+# Alongside OAuth2-Proxy, this file also starts Keycloak to act as the identity provider,
```
Comment (JoelSpeed) [inline contrib/local-environment/docker-compose-traefik.yaml:3]:
```suggestion
# Alongside OAuth2-Proxy, this file also starts Dex to act as the identity provider,
```

File: contrib/local-environment/oauth2-proxy-traefik.cfg [hunk 4]
```diff
@@ -0,0 +1,23 @@
+http_address="0.0.0.0:4180"
+cookie_secret="OQINaROshtE9TcZkNAm-5Zs2Pv3xaWytBmc5W7sPX7w="
+provider="oidc"
+email_domains=["example.com"]
+oidc_issuer_url="http://dex.localhost:4190/dex"
+client_secret="b2F1dGgyLXByb3h5LWNsaWVudC1zZWNyZXQK"
+client_id="oauth2-proxy"
+cookie_secure="false"
+
+# keycloak provider
```
Comment (JoelSpeed) [inline contrib/local-environment/oauth2-proxy-traefik.cfg:10]: Is this still true? Can we double check what's in the normal config and make this look as similar as possible?

**Review by JoelSpeed — COMMENTED** [review]
Couple of bits of the keycloak changes leftover, if we can fix those up then this looks like it should be good to go, I'll give it a try locally to verify it too

File: contrib/local-environment/traefik/traefik.yaml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+api:
+  insecure: true
+log:
+  level: DEBUG
```
Comment (JoelSpeed) [inline contrib/local-environment/traefik/traefik.yaml:4]: Do we want to keep this as debug? I had a play with the environment and the logs were BIG! :sweat_smile: I think info may be better and then users can tweak as appropriate right?

**Comment by fcollonval** [comment]
@JoelSpeed Thanks a lot for the thorough review. I corrected the documentation and switch the log level of Traefik to INFO :wink:

**Review by JoelSpeed — COMMENTED** [review]
Sweet! LGTM, I think this probably deserves a changelog entry, if you can add one, we can get this merged

**Comment by fcollonval** [comment]
> I think this probably deserves a changelog entry

Here it comes

**Review by JoelSpeed — APPROVED** [review]
LGTM, thanks for adding this @fcollonval, I really appreciate having the different examples as this will help many future users who are trying to set OAuth2 Proxy up with Traefik

**Pull request merged**